script.js

```diff
@@ -168,137 +254,121 @@ function addTemp() {
+        // add code
+        return false;
     }
-    if (port2Open === true) {
-        drawPort(wallX, 280);
-    }
-    
-    // Draws active transport
-    drawAT(canvasWidth/2, 50);
-    drawAT(canvasWidth/2, 450);
-}
 
-// Returns a negative or positive number for random movement
-function randomCo() {
-    // Speed + random value so there's always some movement
-    let distance = speed + (Math.random() * Math.floor(speed));
-    
-    // Picks a direction of negative or positive
-    let direction = (Math.random());
-    if (direction > 0.5) {
-        direction = 1;
+    static isCollidingWithAT() {
+        // add code
+        return false;
     }
-    else {
-        direction = -1;
+
+    /************************************/
+
+    updateMolecule() {
+        let tempX, tempY;
+            /* Add x y ranges for checks so lil particles far right/left don't have to go
+            through the checks for the wall/AT and particles high up won't have to go 
+            through checks for lower AT*/
+            
+        do {
+                // Picks a new location for the particle
+                tempX = this.x + randomCo();
+                tempY = this.y + randomCo();
+                // If new location is outside canvas, loop
+        } 
+        while (!isInCanvas(tempX, tempY));
+            
+        // Set particle coordinates to valid new location 
+        this.x = tempX;
+        this.y = tempY;
     }
-    
-    return (distance * direction);
-}
 
-// Checks if the x and y coordinates are inside the canvas
-function isInCanvas(x, y) {
-    if (x > 0 && y > 0 && y < canvasHeight && x < canvasWidth) {
-        return true;
+    // type is passed in because this.moleculeType was not working
+    draw(type) {
+        // Draws Large molecule type
+        if (type == 1) {
+            // Linear gradient for style
+            let gradient = ctx.createLinearGradient(this.x-75, 0, this.x+100, 0);
+            gradient.addColorStop(0, "#540000");
+            gradient.addColorStop(0.5, "red");
+            gradient.addColorStop(1, "#ffac75");
+            
+            drawCircle(this.x, this.y, mole1Radius, gradient);
+        } else {
+            // Draws Small molecule type
+            let gradient = ctx.createRadialGradient(this.x, this.y, mole2Radius*(2/3), this.x, this.y, mole2Radius*(2/15));
+            gradient.addColorStop(0, "#002602");
+            gradient.addColorStop(0.75, "#47772c"); // inside color
+            
+            drawCircle(this.x, this.y, mole2Radius, gradient);
+        }
     }
-    return false;
 }
 
 // Function to fill the coordinate arrays in an orderly fashion
 function particleStartLocations() {
     if (timerOn === false) {
-        // Clear arrays
-        xCo = [];
-        yCo = [];
-        // Big particles
-        xCo = [160, 160];
-        yCo = [125, 375];
-        // Lil particles
-        // R for rows and C for columns
+        // Clear array
+        molecules = [];
```

Comment:
Rather than use separate `xCo[]` and `yCo[]` arrays, we use a single `molecules[]` array which stores `Molecule` instances. The first two are still the two big molecules. Once we figure out how to pass in type properly then it wouldn't matter.